relay-server/src/actors/envelopes.rs

```diff
@@ -1613,11 +1611,12 @@ impl EnvelopeProcessor {
         Ok(())
     }
 
-    #[cfg(feature = "processing")]
     fn filter_event(&self, state: &mut ProcessEnvelopeState) -> Result<(), ProcessingError> {
         let event = match state.event.value_mut() {
             Some(event) => event,
-            None => return Err(ProcessingError::NoEventPayload),
+            // Some events are created by processing relays (e.g. unreal), so they do not yet
+            // exist at this point in non-processing relays.
```

Comment:
Out of scope of this PR, but still: since this was an issue some time ago, and I'm not sure we have strong mechanisms to prevent this from happening again, what do you think if I add a comment after `state.creates_event()`in the `process_state` method? That'll increase visibility in future changes around that part, until we have strong mechanisms in place.

Reply:
Sounds good to me!